### ISSUE TITLE
[Compute contributor wanted] Run the frozen PatchCascade development-selection campaign

### ISSUE THREAD

**Issue by Ayush-Kumar0207** [Ayush-Kumar0207/PatchCascade-SOC#2]
## TL;DR

**What:** independently execute a preregistered RL experiment for cyber-resilience and vulnerability patch management.

**Status:** the reproducibility-hardening phase is complete and development/model-selection compute is authorized. Final training, canonical evaluation, and confirmation evaluation remain sealed.

**Experiment:** Stage 0 compares MultiDiscrete PPO with flattened Discrete + MaskablePPO at equal budget; the mechanical winner then enters the frozen 8→3→2→1 hyperparameter campaign.

**Runtime:** Python 3.11, the exact repository dependency stack, at least 2 GiB free disk at preflight, stable compute, and persistent campaign storage. The committed protocol does not claim a minimum GPU model or VRAM amount.

**Campaign command after qualification:**

```bash
python tools/run_model_selection.py --campaign-dir /persistent/path/patchcascade-selection
```

**You provide:** your compute, persistent storage, and artifact-hosting account. **The repository provides:** both interfaces, candidates, seeds, budgets, five task regimes, evaluator, safety gates, paired statistics, resume identity, ranking, and automatic decisions.

This is an unpaid open-source research contribution. Comment with the qualification details below and wait for acknowledgement before consuming substantial compute.

## Why this is interesting

PatchCascade studies the sequential uptime-versus-security tradeoff in autonomous vulnerability patching across `easy`, `medium`, `hard`, `incident_response`, and `zero_day` regimes. The first scientific question is whether the more complex masked action interface consistently beats the simpler joint-action baseline; the second is which frozen configuration survives safety-first paired selection against a strong heuristic baseline. Negative results remain useful when the evidence is complete.

Repository: https://github.com/Ayush-Kumar0207/PatchCascade-SOC

- [Contributor guide](https://github.com/Ayush-Kumar0207/PatchCascade-SOC/blob/main/TRAINING_CONTRIBUTION_GUIDE.md)
- [Frozen model-selection protocol](https://github.com/Ayush-Kumar0207/PatchCascade-SOC/blob/main/MODEL_SELECTION_PROTOCOL.md)
- [Machine-readable campaign](https://github.com/Ayush-Kumar0207/PatchCascade-SOC/blob/main/training_specs/model_selection_v1.json)

## What you need

- Python 3.11 and exact versions from `requirements-training.txt`
- stable CPU or GPU compute capable of passing repository preflight
- persistent storage outside the source checkout
- your own GitHub, compute, storage, and artifact-hosting credentials; no maintainer credential is shared

The protocol intentionally does not guess a GPU/VRAM requirement or publish an unsupported duration/cost estimate. Please report your actual CPU/GPU, RAM/VRAM, platform, available duration, and interruption constraints. The first qualified contributor will be asked to record wall-clock time by stage, peak memory/VRAM, storage, platform, and interruptions.

## One-command workflow

First reproduce the exact dependency checks, model-selection validator, training preflight, optimizer-shape probe, and six-process/four-worker resume-equivalence proof from the contributor guide. After acknowledgement, run only the campaign command above. Rerun the same command unchanged after interruption; automatic resume must validate the campaign identity.

Do not enter or change interface choices, candidates, seeds, budgets, ranking options, evaluator settings, or survivors.

## What credit you receive

Accepted substantive work receives PR/commit attribution, named Training & Reproducibility Contributors credit, and artifact/model attribution where appropriate. If a manuscript results, acknowledgement or authorship follows actual scholarly contribution and applicable authorship standards. Compute alone never guarantees authorship.

## How to claim

Comment once with:

- CPU/GPU model, RAM/VRAM, platform, OS/runtime image, Python constraints, and available duration;
- persistent-storage plan and expected interruption/resume pattern (do not post private local paths, credentials, or account identifiers);
- artifact host, retention plan, and ability to publish immutable links plus SHA-256 hashes;
- whether you want Track A, B, C, or D below;
- confirmation that the contribution is unpaid and uses only your own accounts.

Wait for a public acknowledgement before starting substantial compute so full campaigns are not duplicated.

## Participation tracks

- **Track A — environment/preflight verifier:** reproduce exact dependency, environment, optimizer-shape, and resume-equivalence checks; small commitment, no campaign-result claim.
- **Track B — development compute contributor:** run interface selection followed by the frozen 8→3→2→1 campaign and return the complete evidence directory.
- **Track C — independent artifact/reproducibility reviewer:** verify paired-interface evidence, identities, completeness, resume lineage, gates, and decisions without choosing results.
- **Track D — research contributor:** contribute substantial analysis, methodology, or writing through a separate reviewed research change.

Tracks A–C do not carry promised authorship.

## Detailed integrity requirements

- Use the exact acknowledged source commit and a clean checkout.
- Stop on every `STOP:` error; do not repair result files or change the run in place.
- Preserve the complete campaign directory, `campaign_lock.json`, `campaign_state.json`, interface records, `interface_decision.json`, candidate/seed/task records, round decisions, `selection_decision.json`, logs, checkpoints, manifests, and `proposed_final_spec.json`.
- Never rerun only an unfavorable episode, delete evidence, substitute a checkpoint, or manually override an interface or survivor.
- Never run final training, canonical, or confirmation work during this phase.
- Treat runtime `.pkl` sidecars as executable trusted-run material; never ask a maintainer to resume an unaudited downloaded checkpoint.
- Do not put model binaries, checkpoints, tokens, `.env` files, or private runtime material in Git.

The full fail-closed process remains authoritative in the contributor guide and committed specification.
